src/track/beatutils.cpp

```diff
@@ -5,44 +5,350 @@
+        // The analyzer has some jitter that causes a steady beat to fluctuate around the correct
+        // value so we don't consider changes to a neighboor value at the ordered tempo table
+        if (newStableTempo == stableTemposByPosition.last()) {
+            continue;
+            // Here we check if the new tempo is the right neighboor of the previous tempo
+        } else if (stableTemposByPosition.last() != frequencyOfTempos.lastKey() and
+                newStableTempo == (frequencyOfTempos.find(stableTemposByPosition.last()) + 1).key()) {
+            continue;
+            // Here we check if the new tempo is the left neighboor of the previous tempo
+        } else if (stableTemposByPosition.last() != frequencyOfTempos.firstKey() and
+                newStableTempo == (frequencyOfTempos.find(stableTemposByPosition.last()) - 1).key()) {
+            continue;
+        } else {
+            // This may not be case when our median window is even, we can't use it
+            // Because find() will return an iterator pointing to end that we will *
+            if (frequencyOfTempos.contains(newStableTempo)) {
+                lastBeatChange = currentBeat - tempoMedianFilter.lag() - mostFrequentTempo.lag();
+                // Since we used the median as guess of the first tempo
+                // we change position at 0 if we detect a change in very beginning 
+                if (lastBeatChange > kBeatsToCountTempo) {
+                    stableTemposByPosition[lastBeatChange] = newStableTempo;
+                } else {
+                    stableTemposByPosition[0] = newStableTempo;
+                }
+            }
+        }
+    }
+    // We also add the median as rough guess as our last tempo
+    // This way we always have both sentinels for the whole track
+    // as a constant tempo region if no significant tempo changes
+    stableTemposByPosition[tempoList.count() - 1] = medianBpm;
+    qDebug() << stableTemposByPosition;
+    return stableTemposByPosition;
+}
+
+void BeatUtils::removeSmallArrhythmic(
+        QVector<double>& beats, int sampleRate, QMap<int, double>& stableTemposByPosition) {
+    // A common problem the analyzer has is to detect arrhythmic regions
+    // of tacks with a constant tempo as in a different unsteady tempo.
+    // This happens frequently on builds and breaks with heavy effects on edm music.
+    // Since these occurs most on beatless regions we do not want them to be
+    // on a different tempo, because they are still syncable in the true tempo
+    // We arbitraly remove these arrhythmic regions if they are short than 16s
+    // and return to same tempo they start deviating from.
+    // 16s is 32 beats or 8 measure on a typical 4/4 120 bpm which is a resonable
+    // guess for a intented musical change
+    // TODO(Cristiano) Use a better heuristic for "finding" these regions
+    // like for example the avarage energy of the beats
```

Comment:
```suggestion
    // A common problem the analyzer encounters is to detect the tempo in regions without
    // prominent rhythmic sounds. This happens frequently on builds and breaks
    // with heavy effects in EDM. These occur mostly on beatless regions,
    // so we do not want them to be marked as a different tempo because they are
    // still syncable in the true tempo. We remove these arrhythmic regions if they are shorter
    // than a 16s threshold and return to same tempo they start deviating from.
    // 16s is 32 beats or 8 measure on a typical 4/4 120 bpm which is a reasonable
    // guess for an intended musical change.
    // TODO(Cristiano) Use a better heuristic for "finding" these regions
    // like for example the average energy of the beats.
```

Reply:
Should the threshold be calculated using the tempo instead of an arbitrary assumption of 120 BPM?